modules/ml/src/main/java/org/apache/ignite/ml/naivebayes/bernoulli/BernoulliNaiveBayesTrainer.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.ml.naivebayes.bernoulli;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.ignite.ml.dataset.Dataset;
+import org.apache.ignite.ml.dataset.DatasetBuilder;
+import org.apache.ignite.ml.dataset.UpstreamEntry;
+import org.apache.ignite.ml.dataset.primitive.context.EmptyContext;
+import org.apache.ignite.ml.math.functions.IgniteBiFunction;
+import org.apache.ignite.ml.math.primitives.vector.Vector;
+import org.apache.ignite.ml.trainers.SingleLabelDatasetTrainer;
+
+/**
+ * Trainer for the Bernoulli naive Bayes classification model. The trainer calculates prior probabilities from the input
+ * binary dataset. Prior probabilities can be also set by {@code setPriorProbabilities} or {@code
+ * withEquiprobableClasses}. If {@code equiprobableClasses} is set, the probalilities of all classes will be {@code
+ * 1/k}, where {@code k} is classes count.
+ */
+public class BernoulliNaiveBayesTrainer extends SingleLabelDatasetTrainer<BernoulliNaiveBayesModel> {
```

Comment:
Formally Bernoulli distribution describes discrete distribution for two values (classes), but you have created trainer for more general case. From this point of view class name should be changed to "DiscreteNaiveBayes" for example

Reply:
@dehasi Excuse me for my previous comment. I was wrong and I hurried. Please review my new comment above.

Reply:
There is a [task](https://issues.apache.org/jira/browse/IGNITE-9745) to add Multinomial Naive Bayes where I'm going to implement a more general trainer.
It supposed both this and IGNITE-9745 be subtasks for IGNITE-9979. But I wasn't able to move them.